tests/test_int32.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# pylint: disable=missing-docstring
+import unittest
+
+import hypothesis
+import hypothesis.strategies
+
+import numenc_module
+
+MIN_SIGNED_INT32 = -2147483648
+MAX_SIGNED_INT32 = 2147483647
+MAX_UNSIGNED_INT32 = 4294967295
+
+
+class TestEncodingInt32(unittest.TestCase):
+    @hypothesis.given(hypothesis.strategies.integers(min_value=MIN_SIGNED_INT32, max_value=MAX_SIGNED_INT32))
+    def test_encode_decode_automatic(self, value: int):
+        hypothesis.assume(value)
+        self.assertEqual(value, numenc_module.to_int32(numenc_module.from_int32(value)))
+
+    @hypothesis.given(
+        value1=hypothesis.strategies.integers(min_value=MIN_SIGNED_INT32, max_value=MAX_SIGNED_INT32),
+        value2=hypothesis.strategies.integers(min_value=MIN_SIGNED_INT32, max_value=MAX_SIGNED_INT32))
+    def test_encode_order(self, value1: int, value2: int):
+        encoded1 = numenc_module.from_int32(value1)
+        encoded2 = numenc_module.from_int32(value2)
+        if value1 < value2:
+            self.assertLess(encoded1, encoded2)
+        elif value1 == value2:
+            self.assertEqual(encoded1, encoded2)
+        else:
+            self.assertGreater(encoded1, encoded2)
+
+    def test_encode_decode(self):
+        #yapf: disable
+        nums = [
+            MIN_SIGNED_INT32,
+            -333332,
+            0,
+            2221289,
+            MAX_SIGNED_INT32
+        ]
+
+        expected = [
+            b'\x00\x00\x00\x00',
+            b'\x7f\xfa\xe9\xec',
+            b'\x80\x00\x00\x00',
+            b'\x80\x21\xe4\xe9',
+            b'\xff\xff\xff\xff'
+        ]
+        #yapf: enable
+
+        for i, num in enumerate(nums):
+            key = numenc_module.from_int32(num)
+            self.assertEqual(expected[i], key)
+
+            num_decoded = numenc_module.to_int32(key)
+            self.assertEqual(num, num_decoded)
+
+        for i, _ in enumerate(expected):
+            if i > 0:
+                self.assertLess(expected[i - 1], expected[i])
+
+    def test_encode_exceptions(self):
+        tajp_err_triggers = [
+            "some string", 2.344, ('1', '2'), [], {}, b'\x01\x02', MIN_SIGNED_INT32 - 1, MAX_SIGNED_INT32 + 1
+        ]
+
+        for weird_val in tajp_err_triggers:
+            tajp_err = False
+            try:
+                _ = numenc_module.from_int32(weird_val)
+            except TypeError:
+                tajp_err = True
+
+            self.assertTrue(tajp_err, msg="excepted error thrown for input {!r}, but got no error.".format(weird_val))
```

Comment:
I would check for the error message:

```python
self.assertIsNotNone(type_err, msg="Expected error to be thrown for input ...")
self.assertEqual('some error message', str(type_err))
```

Please also change all the test cases above/below.